### PR TITLE
Fixed spectral dependence of eval_sigma_t_mono

### PR DESCRIPTION
**Pull request by nollety**
Resolves eradiate/eradiate-issues#106

# Description

Fixed `ParticleLayer.eval_sigma_t_mono()`.

# Checklist

- [x] The code follows the relevant coding guidelines
- [x] The code generates no new warnings
- [x] The code is appropriately documented
- [x] The code is tested to prove its function
- [x] The feature branch is rebased on the current state of the `main` branch
- [x] I give permission that the Eradiate project may redistribute my contributions under the terms of its license

### CODE REVIEW

File: eradiate/scenes/atmosphere/_particles.py [hunk 1]
```diff
@@ -370,18 +371,21 @@ def eval_sigma_t(self, spectral_ctx: SpectralContext) -> pint.Quantity:
...
 
         ds = xr.open_dataset(self.dataset)
-        wavelengths = w.m_as(ds.w.attrs["units"])
-        interpolated_sigma_t = ds.sigma_t.interp(w=wavelengths)
-        sigma_t = to_quantity(interpolated_sigma_t)
+        ds_w_units = ds.w.attrs["units"]
+        wavelength = w.m_as(ds_w_units)
+        sigma_t = to_quantity(ds.sigma_t.interp(w=wavelength))
+        sigma_t_550 = to_quantity(
+            ds.sigma_t.interp(w=ureg.Quantity(550.0, "nm").m_as(ds_w_units))
```
Comment (leroyvn) [inline eradiate/scenes/atmosphere/_particles.py:378]: For optimal performance, I'd rather rather go with
```suggestion
            ds.sigma_t.interp(w=ureg.convert(550.0, ureg.nm, ds_w_units))
```

File: eradiate/scenes/atmosphere/_particles.py [hunk 2]
```diff
@@ -370,18 +371,21 @@ def eval_sigma_t(self, spectral_ctx: SpectralContext) -> pint.Quantity:
     def eval_sigma_t_mono(self, w: pint.Quantity) -> pint.Quantity:
 
         ds = xr.open_dataset(self.dataset)
-        wavelengths = w.m_as(ds.w.attrs["units"])
-        interpolated_sigma_t = ds.sigma_t.interp(w=wavelengths)
-        sigma_t = to_quantity(interpolated_sigma_t)
+        ds_w_units = ds.w.attrs["units"]
```
Comment (leroyvn) [inline eradiate/scenes/atmosphere/_particles.py:374]: How about pre-computing those:
```suggestion
        ds_w_units = ureg.Unit(ds.w.attrs["units"])
```

**Review by leroyvn — APPROVED** [review]
Hi @nollety, just a couple of minor remarks. Please be sure to double check my suggestions. Looks good to me apart from that.

**Comment by nollety** [comment]
@leroyvn Thanks for your suggestions, they look good to me! :smiley

**Comment by nollety** [comment]
@leroyvn ready for merging!

**Pull request merged**